### PR TITLE
Fix inconsistent locations in some resources - Part II

### PR DESCRIPTION
**Pull request by upodroid**
@melinath

Part of https://github.com/GoogleCloudPlatform/magic-modules/pull/4377

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
memcache: fixed an issue where `google_memcached_instance` resource was not inferring the region from the provider.
```
```release-note:enhancement
vpcaccess: fixed an issue where `google_vpc_access_connector` resource was not inferring the region from the provider.
```
```release-note:enhancement
tpu: fixed an issue where `google_tpu_node` resource was not inferring the zone from the provider.
```

### CODE REVIEW

File: templates/terraform/examples/memcache_instance_basic.tf.erb [hunk 1]
```diff
@@ -22,7 +22,6 @@ resource "google_service_networking_connection" "private_service_connection" {
 resource "google_memcache_instance" "<%= ctx[:primary_resource_id] %>" {
   provider = google-beta
   name = "<%= ctx[:vars]["instance_name"] %>"
-  region = "us-central1"
```
Comment (melinath) [inline templates/terraform/examples/memcache_instance_basic.tf.erb:25]: Based on Riley's comments on the other PR, I would lean towards keeping `us-central1` here and for vpc access connector, unless there's a particular reason for removing them?

**Review by melinath — CHANGES_REQUESTED** [review]
These changes seem reasonable to me. Just have one question about the test examples.

**Comment by melinath** [comment]
Also, could the release note be updated to be per-product? Specifically, three separate code blocks, one for each product being altered. That'll make it be displayed correctly in the downstream changelog. Thanks!

File: templates/terraform/examples/memcache_instance_basic.tf.erb (hunk 1 above)
Comment (upodroid) [reply templates/terraform/examples/memcache_instance_basic.tf.erb:25]: Removing that infers the region from the provider. All tests run us-central1 where as the zone defaults to us-central1-a and that needed to be changed to be b

**Comment by upodroid** [comment]
fixed the release notes

File: templates/terraform/examples/memcache_instance_basic.tf.erb (hunk 1 above)
Comment (melinath) [reply templates/terraform/examples/memcache_instance_basic.tf.erb:25]: 👍 so they do. Thanks!

**Review by melinath — APPROVED** [review]
LGTM - thanks!

**Pull request merged**